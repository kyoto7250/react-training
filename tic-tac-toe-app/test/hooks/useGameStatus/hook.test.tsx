import {
    useGameStatus
} from '../../../src/hooks/useGameStatus/hook'
import { renderHook } from '@testing-library/react-hooks'



describe('useGameStatus', () => {
    let result: any;

    beforeEach(() => {
        result = renderHook(() => useGameStatus()).result
    })

    test('check init value', () => {
        expect(result.current.board).toMatchSnapshot()
        expect(result.current.sentence).toBe("Next player: X")
        expect(<>{result.current.moves}</>).toMatchSnapshot()
    })
})

