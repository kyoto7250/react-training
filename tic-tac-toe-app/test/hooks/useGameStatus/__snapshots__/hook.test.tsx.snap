// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useGameStatus check init value 1`] = `
<Board
  onClick={[Function]}
  squares={
    Array [
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ]
  }
/>
`;

exports[`useGameStatus check init value 2`] = `
<React.Fragment>
  <li>
    <button
      onClick={[Function]}
    >
      Go to game start
    </button>
  </li>
</React.Fragment>
`;
